jdaviz/configs/imviz/plugins/reproject/reproject.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import time
+from glue.core.data import Data
+from traitlets import Bool, Unicode
+
+from jdaviz.core.events import SnackbarMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, DatasetSelectMixin
+
+try:
+    from reproject import reproject_interp
+    from reproject.mosaicking import find_optimal_celestial_wcs
+except ImportError:
+    HAS_REPROJECT = False
+else:
+    HAS_REPROJECT = True
+
+__all__ = ['Reproject']
+
+
+@tray_registry('imviz-reproject', label="Reproject")
+class Reproject(PluginTemplateMixin, DatasetSelectMixin):
+    template_file = __file__, "reproject.vue"
+
+    results_label = Unicode("Reprojected").tag(sync=True)
```

Comment:
I thought about using the mixin class but the "add result to viewer" over there is too flexible to my liking, so I opted to keep this simple for now.

https://github.com/spacetelescope/jdaviz/blob/95c34c2e5eaffd76352cfea2e93e6501c9c433c2/jdaviz/core/template_mixin.py#L1586

Reply:
what is too flexible about it?  Maybe we need a subclass that is less flexible or just options to make it do what you want here?

Reply:
I want to be able to control exactly which viewer it is going to and then do more stuff to that viewer, so I cannot use any of the viewer logic in the mixin but I also don't know how to turn it off over there.

Reply:
If you can't set the filter to only resolve to the viewer you want, then we should add that capability so that you can.  Once that is in place, the dropdown will become a switch and you'll still get all the consistent labeling logic used in all the other plugins.

Reply:
all other plugins use a switch (so that you can add the results to the data collection, but not to a viewer), but if we really don't want that here, we can add an option to hide the switch as well.  I just think we should try to use these components so that behavior does not diverge between plugins.

Reply:
then let's either hide the switch, force it to be on, or use the `AutoTextField` component for the label instead and just live with some code duplication?

Reply:
All those SGTM, which one would have the least code impact?

Reply:
After talking, I think the `AutoTextField` will be sufficient.  That will let you set the default label to be based on the input data-label (perhaps appending `(reprojected)`) and you can easily add validation errors if its invalid or overwriting.  If you want to support overwriting an existing label, then I think it makes more sense to use the `AddResults` component and am happy to help hide the switch.